### PR TITLE
Support external goma client and make MacOS goma-friendly (uplift to 1.35.x)

### PR DESCRIPTION
**Pull request by goodov**
Uplift of #11838
Resolves https://github.com/brave/brave-browser/issues/20347

We don't have `isBraveReleaseBuild()` flag in `1.35.x` so I use `!shouldSign()` to disable dSYMs in this branch (PR builds are unsigned). True `Release` builds are signed, in this case dSYMs will be active as intended.

Pre-approval checklist: 
- [x] You have tested your change on Nightly. 
- [ ] This contains text which needs to be translated. 
    - [ ] There are more than 7 days before the release. 
    - [ ] I've notified folks in #l10n on Slack that translations are needed. 
- [ ] The PR milestones match the branch they are landing to. 

Pre-merge checklist: 
- [x] You have checked CI and the builds, lint, and tests all pass or are not related to your PR. 

Post-merge checklist: 
- [x] The associated issue milestone is set to the smallest version that the changes is landed on.

### CODE REVIEW

**Review by kjozwiak — APPROVED** [review]
Uplift into `1.35.x` approved after deliberating with @brave/uplift-approvers. @goodov confirmed this was verified/checked on Nightly.

File: build/commands/lib/config.js
Comment (bridiver) [inline build/commands/lib/config.js:296]: not sure how `!this.shouldSign()` ended up in the uplift because it was removed from the original PR because it's not correct and is causing problems https://bravesoftware.slack.com/archives/C7VLGSR55/p1643325085370429?thread_ts=1643323803.539619&cid=C7VLGSR55

Comment (bsclifton) [reply build/commands/lib/config.js:296]: Not sure how this got here either - but just wanted to share that after discussion, this was addressed in https://github.com/brave/brave-core/pull/12040

**Pull request merged**
